cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java

```diff
@@ -163,9 +147,30 @@ private void checkKafkaTopic(Set<String> problemKeys) {
     }
   }
 
-  private boolean isPositiveInteger(String key) {
+  private boolean verifyResources(ServiceResourceKeys serviceResourceKeys, Set<String> problemKeys) {
     try {
-      return cConf.getInt(key) > 0;
+      boolean allPositive = true;
+      if (serviceResourceKeys.getMemory() <= 0) {
+        LOG.error("  {} must be a positive integer", serviceResourceKeys.getMemoryKey());
```

Comment:
ServiceResourceKey objects are initialized with the key names from Constants so getMemoryKey() returns the actual key.